### ISSUE TITLE
BIP not implemented on docker for windows containers?

### ISSUE THREAD

**Issue by BarnumD** [docker/for-win#118]
Using Windows containers, adding bip to the docker daemon doesn't change the container's IP like it does on docker for windows with Linux containers.
`"bip": "10.11.12.1/23",`
The ip address is something other than 10.11.12.x

**Comment by BarnumD** [comment]
Thanks, but, for me that's not a fix :(  We rely on that because docker chooses networks that are part of our corporate ranges.

**Comment by dgageot** [comment]
@BarnumD No worries, we are going to both fix the GUI and add the feature. Sorry for the misunderstanding.

**Comment by rneugeba** [comment]
@BarnumD for now you should be able to change the IP prefix used manually as documented [here](https://msdn.microsoft.com/en-gb/virtualization/windowscontainers/management/container_networking).

**Comment by dgageot** [comment]
@BarnumD This is scheduled to be released in next version

**Comment by rneugeba** [comment]
@dgageot this does not seem to work with `1.13.0-rc4-beta34 (9562)`:
```
PS C:\Users\rneugeba> cat C:\ProgramData\Docker\config\daemon.json
{  "fixed-cidr": "10.11.12.1/23"}
PS C:\Users\rneugeba> docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
d48a87f2c829        nat                 nat                 local
2cdd1b1a909b        none                null                local
PS C:\Users\rneugeba> docker network inspect nat
[
    {
        "Name": "nat",
        "Id": "d48a87f2c82917879b51de57e26a9ccef801014d0f7668f9f3778d50be8dd5c2",
        "Created": "2016-12-29T18:47:13.9940174Z",
        "Scope": "local",
        "Driver": "nat",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "windows",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.26.64.0/20",
                    "Gateway": "172.26.64.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Containers": {},
        "Options": {
            "com.docker.network.windowsshim.hnsid": "718eff6f-b73a-48fd-9871-ce87be1a2117",
            "com.docker.network.windowsshim.networkname": "nat"
        },
        "Labels": {}
    }
]
PS C:\Users\rneugeba> docker version
Client:
 Version:      1.13.0-rc4
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   88862e7
 Built:        Sat Dec 17 01:34:17 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.0-rc4
 API version:  1.25 (minimum version 1.24)
 Go version:   go1.7.3
 Git commit:   88862e7
 Built:        Sat Dec 17 01:34:17 2016
 OS/Arch:      windows/amd64
 Experimental: true
```

using `bip` gives an error because it's not a windows Daemon option, but https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/container-networking suggests to use `fixed-cidr`. Any idea?

**Comment by rneugeba** [comment]
I'm closing this issue. We've added the option to change the daemon config to the UI a while back. If the settings in the daemon config don't work, this would be an issue with the docker engine itself and an issue should be filed on https://github.com/docker/docker.
thanks for the report

**Comment by rubin55** [comment]
This actually does not work, and is not clearly documented:

1: When Docker switches-to/uses Windows Containers on Windows, the assigned address is controlled by something that I *think* exists outside of docker. You can see that configuration by running "Get-NetNAT" in PowerShell.

2: When Docker switches-to/uses Linux Containers on Windows, the assigned address comes from the docker0 bridge configuration which you can see by running "docker network inspect bridge". This network is not removable, and its settings cannot be controlled by --bip and --gateway settings, since these don't work on Windows (as documented on docs.docker.com). These settings also do not work when specified in daemon.json.

3: The GUI for Docker for Windows has a Network tab that shows only a network range and a dns server address. However, the address configured there seems to be the address range for something that performs NAT services, ie: HOST <-> 10.0.75.1 <-> 172.17.0.x <-> CONTAINER. 

Ergo, I don't know HOW someone would change the network range given to running containers by connecting to the default bridge on Windows.

**Comment by fabricek** [comment]
Completing C:\Users\username\.docker\daemon.json file with bip and fixed-cidr works for me. Something like { ...."experimental": true, "bip": "192.168.2.192/25", "fixed-cidr": "192.168.0.224/27" } in the daemon.json file and my containers are created on 192.168.2xx network after restarting docker for windows.

**Comment by jigarakatidus** [comment]
For Windows 7 - Docker Toolbox, [this](https://stackoverflow.com/a/43629665/861704) solution worked.

**Comment by ddzobov** [comment]
> { ...."experimental": true, "bip": "192.168.2.192/25", "fixed-cidr": "192.168.0.224/27" }

I use latest docker for windows - daemon.json not working for me, containers still created on 172.18 network
